src/components/modals/AddressSelectorModal.vue

```diff
@@ -32,17 +32,28 @@ export default defineComponent({
     setup(props, context) {
         const { activeCurrency } = useAccountStore();
 
-        const { path } = context.root.$router.currentRoute;
+        const { name } = context.root.$router.currentRoute;
 
         function addressSelected() {
             disableNextModalTransition();
 
-            context.root.$router.push(`${path}/${activeCurrency.value}`);
+            context.root.$router.push({
+                name: `${name}-${activeCurrency.value}`,
+                params: {
+                    canUserGoBack: 'true',
```

Comment:
Does `true` have to be a string?

Reply:
You are right, adding a 'false' as a string will have the same behaviour, although I've added a comment 😸